0006/README.md

```diff
@@ -0,0 +1,319 @@
+# `docs/spec/v0.1/*`,
+# `docs/provenance/v0.1/*`,
+# `docs/verification_summary/v0.1/*`
+slsa-framework/slsa:release/v0.1
+├─ images/     # contains images specifically for this version.
+└─ <v0.1-versioned-files>.md
+```
+
+#### Main Branch
+
+The directory structure of the `main` branch stays the same as the repo has
+right now.
+
+*What is changing*
+
+-   `/docs/spec`, `/docs/provenance`, `/docs/verification_summary`, and
+    `/docs/github-actions-workflow` are removed from the main branch.
+-   `/docs/images` doesn't contain versioned files and directories anymore. They
+    are removed or moved to their branch.
+-   `/docs/v1.0` and `/docs/v0.1` are two submodules that point to the
+    [version branches](#version-branches). As a result, `.gitmodules` is added
+    in the top level directory, and has `url = ./` for each submodule config to
+    ensure referencing to the local repo. A future version should follow a
+    similar setup in the repo.
+-   All permalinks in file content use the dir structure described in
+    [the website](#the-website) section.
+
+NOTE: We admit the way of using the Git submodule is non-standard, but it is a
+nice compromise and helps us to improve the maintenance experience.
+
+#### Version Branches
+
+Git branch is the way to manage versioned specifications.
+
+*What is changing*
+
+-   `release/<version|version-draft>` is the branch pattern for a prior or
+    future `version`, where the `version` must contain both major and minor.
+-   In addition to the shared images in the `main` branch, if an image is
+    required for a specific version, the `/images` directory in the version
+    branch may contain the image and let the versioned files reference it .
+-   At the top level directory, a branch has all the versioned file from
+    `/docs/spec/<version>/*`, `/docs/provenance/<version>.md`,
+    `/docs/verification_summary/<version>.md`, and
+    `/docs/github-actions-workflow/<version>.md`. All specification files in the
+    same branch share the same version.
+
+## Automate Building and Deploying The Website
+
+A shell script should assemble the website, instead of using any logic in the
+Github Workflow script.
+
+### Checkout repo, load all submodules, and build once
+
+The GitHub Workflow clones the `main` branch once, pulls the latest files for
+all the submodules, builds the website once, and deploys the website to Github
+Pages.
+
+*Disadvantages*
+
+-   The usage of Git submodule is non-standard, and it leaves some versioned
+    directories in the `main` branch.
+
+*What is changing and code snapshot*
+
+The following is a snapshot of the scripts for building and deploying the
+website.
+
+-   To serve the website locally,
+    [/docs/README.md](https://github.com/slsa-framework/slsa/blob/main/docs/README.md)
+    is replaced with running `build.sh` or similar logic.
+-   The Github Workflow uses the same shell script to prepare the build
+    environment.
+    
+```
```

Comment:
nit: I'm not sure we really need a script here. The only difference from the existing instructions is how to checkout (using submodules). Otherwise it's identical, right? That would be good to explain, to show how it's not a big change.

I have comments about the github config, but I'll put that as a separate comment.

(If you keep it, use ```` ```bash ```` to enable syntax highlighting.)